cpp/src/arrow/util/tdigest.h

```diff
@@ -60,12 +61,27 @@ class ARROW_EXPORT TDigest {
     input_.push_back(value);
   }
 
+  // skip NAN on adding
+  // TODO(yibo): store NAN as is, partition to buffer end before merging
+  template <typename T>
+  typename std::enable_if<std::is_floating_point<T>::value>::type NanAdd(T value) {
+    if (!std::isnan(value)) Add(value);
+  }
+
+  template <typename T>
+  typename std::enable_if<std::is_integral<T>::value>::type NanAdd(T value) {
+    Add(static_cast<double>(value));
```

Comment:
Note that `std::isnan` is defined for integrals. So you could probably just write:
```c++
  template <typename T>
  void NanAdd(T value) {
    if (!std::isnan(value)) { Add(static_cast<double>(value)); }
  }
```

Reply:
But looks MSVC doesn't support it? https://godbolt.org/z/h6jxGo

Reply:
Ah, too bad.